services/datamanager/datasync/sync.go

```diff
@@ -219,9 +216,9 @@ func (s *syncer) unmarkInProgress(path string) {
 	delete(s.inProgress, path)
 }
 
-// exponentialRetry calls fn, logs any errors, and retries with exponentially increasing waits from initialWait to a
+// exponentialRetry calls fn and retries with exponentially increasing waits from initialWait to a
 // maximum of maxRetryInterval.
-func exponentialRetry(cancelCtx context.Context, fn func(cancelCtx context.Context) error, log golog.Logger) error {
+func exponentialRetry(cancelCtx context.Context, fn func(cancelCtx context.Context) error) error {
```

Comment:
Instead of moving logging out of this function for better observability into what's going on, could we just pass the file name into this function? Asking since if a file never fails to get uploaded due to some unchanging reason, it'll continuously keep failing to upload without notifying the user, since I believe we only return an error here if the context is canceled.

Reply:
Bump on this question.

Reply:
Sorry totally missed this! Great call out, yeah we'd definitely want to be made aware of that. I wanted to keep exponentialRetry generic, so went ahead and added the logging to the inline definition of `fn` where `exponentialRetry` is called